doc/develop/tools/clion.rst

```diff
@@ -0,0 +1,267 @@
+     the path to your flashing software (OpenOCD in our example).
+
+#. Save the file. Make sure it is located in the ``app`` folder.
+
+
+Load CMake
+**********
+
+#. In the **Project** tree, locate the ``CMakeLists.txt`` file in the ``app``
+   folder.
+
+#. Right-click it and select **Load CMake Project**.
+
+   CMake load should finish successfully.
+
+
+Configure CMake profiles
+************************
+
+#. Go to **Settings | Build, Execution, Deployment | CMake**.
+
+#. Delete the automatically created **Debug** profile.
+
+#. Select the **Zephyr-Debug preset** profile and set the **Enable profile**
+   checkbox.
+
+#. Click **Apply** and close the dialog.
+
+   CMake project will be reloaded.
+
+
+Build the project
+*****************
+
+#. Check the top toolbar: the **Zephyr-Debug** profile and the
+   **zephyr_final** configuration should be selected.
+
+#. Click the hammer icon next to the configuration switcher.
+
+   When the build is finished, you can find build artifacts in
+   the ``app\build`` folder.
+
+
+Configure Zephyr parameters for debug
+*************************************
+
+#. In the configuration switcher, select **guiconfig** and click the hammer
+   icon.
+
+#. Use the GUI application to set the following flags:
+
+   .. code-block::
+
+      CONFIG_THREAD_MONITOR
+      CONFIG_THREAD_NAME
+      CONFIG_THREAD_RUNTIME_STATS
+      CONFIG_DEBUG_OPTIMIZATIONS
+
+Enable RTOS integration
+***********************
+
+#. Go to **Settings | Build, Execution, Deployment | Embedded Development |
+   RTOS Integration**.
+
+#. Set the **Enable RTOS Integration** checkbox.
+
+   This option enables Zephyr tasks view during debugging. See
+   `Multi-threaded RTOS debug`_ in CLion web help for more information.
+
+   .. _Multi-threaded RTOS debug: https://www.jetbrains.com/help/clion/rtos-debug.html
+
+
+Create an Embedded GDB Server configuration
+*******************************************
+
+#. In the configuration switcher, select **Edit Configurations**.
+
+#. Click "+" in the left-hand pane and select **Embedded GDB Server**.
+
+#. Configure the settings:
+
+   * **Name** (optional): Debug Zephyr
+
+   * **Target**: zephyr_final
+
+   * **Executable binary**: zephyr_final
+
+   * **Debugger**: Bundled GDB
+
+      *Note*: For non-ARM and non-x86 architectures, use a GDB executable form
+      Zephyr SDK. Make sure to pick a version with Python support (for
+      example, **riscv64-zephyr-elf-gdb-py**) and check that Python is
+      presented in the system ``PATH``.
+
+   * **Download executable** (optional): specify when to download the
+      executable binary to the target device.
+
+   * **'target remote' args** (optional): provide the GDB server-specific
+      arguments (for example, port number or board config file).
+
+   * **GDB Server**: cmake
+
+   * **GDB Server args**:
+
+      ``--build --preset Zephyr-Debug --target flash debugserver``
+
+   * **Working directory**: the path to ``app``
```

Comment:
```suggestion
   * **Working directory**: Directory where `CMakeUserSettings.json` is located so that CMake can find the Zephyr-Debug CMake preset, i.e. ``app``.
```

Also, could this be set using a macro (clicking the "+") instead? having to navigate to the folder and set it as an absolute path seems wrong? `$CMakeCurrentBuildDir$/..` works, but not sure if there is a better one?

Reply:
Unfortunately, some macros can be buggy, so it's safer to ask for the full path